### PR TITLE
Show protest card on marker popup

### LINKED ISSUE #14: Popup with demonstration info in general map when clicking on a marker

### PR DESCRIPTION
**Pull request by oori**
Resolves https://github.com/guytepper/1km.co.il/issues/14

### CODE REVIEW

**Comment by guytepper** [comment]
Wow, that's amazing. Thank you! Our designer loved it too!

Seems like it doesn't work well on mobile. Can you disable it there for the time being?
Also, take a look at the `PortestCard` styles, as some of them gets overridden from the popup (e.g. font)

**Review by uriklar — APPROVED** [review]

**Comment by oori** [comment]
@guytepper it does work on mobile, seems the issue in only on Safari, unrelated to platform, as it does work on chrome-iOS and firefox-iOS, for example.   I'll check it now (and test Androids as well).
Regarding the style,  leaflet styles need to be overridden.   for now, I'll just to it manually (`!important` on specific attributes), but we should probably get rid of those `!important` and fix it it proper css flow, after you do the redesign.

**Comment by oori** [comment]
@guytepper I've validated on a couple of platforms, the issue is **only** in Safari.   Unfortunately, it's hard to debug Safari due to a couple of stuff (_mainly: geoloc require HTTPS on localhost, app doesn't work without.._).    I will take care for this later today.

**Comment by oori** [comment]
The safari issue was always there, in your **existing online** version as well. Unrelated to this PR.
It originates because only on safari (_why??_), when you click a `Marker`, it panes the map to center the `Marker`.  so [onMoveEnd](https://github.com/guytepper/1km.co.il/blob/62a8704ed1b163769a1fb9e4897411411a1936cd/src/components/Map/Map.js#L52) event calls `setMapPosition`, the state changes, and you get in react in a rendering loop.

I'm looking if this "click-to-pane" can be bypassed for safari. surprisingly, it's not in the api, nor google.. maybe related to new safari.

EDIT:  it's a bug in leaflet 1.7.   see: [issue](https://github.com/Leaflet/Leaflet/issues/7255)

**Comment by oori** [comment]
@guytepper after testing a bunch of solutions/hacks (`debounce`, `useMemo`, ...), the best and **cleanest** solution is to downgrade back to leaflet 1.6.     I've tested it now, adding to this PR.   ~~Unfortunately, Vercel doesn't rebuild it (link still runs leaflet 1.7.1 `L.version`).~~ _(EDIT: oops.. fixed yarn.lock, [vercel link](https://1km-qofblp5s4.vercel.app) now correct)_
I'll keep tracking the leaflet issue, hope this fix is soon..  it affect all leaflet-marker-safari.

File: package.json [hunk 1]
```diff
@@ -8,7 +8,7 @@
     "firebase": "^7.21.1",
     "geofirestore": "^4.3.0",
     "geolib": "^3.3.1",
-    "leaflet": "^1.7.1",
+    "leaflet": "1.6.0",
```
Comment (uriklar) [inline package.json:11]: @oori Why was this required?

Comment (oori) [reply package.json:11]: @uriklar Leaflet 1.7 has an issue Safari (on all platforms), it sends duplicate events (like `onMoveEnd`), which makes mess. See my comment above (`debounce`, etc..), also note it's unrelated to my PR, it exists on online site and sometimes crashes Safari randomly on map clicks (_react state changes timing_), you just didn't get enough complaints...   There are a couple of open issues on leaflet, see: https://github.com/Leaflet/Leaflet/issues/7255,  reverting to leaflet 1.6 is the safest solution for now.

**Comment by guytepper** [comment]
Hi @oori, sorry for taking so long to check this out.
When I press multiple times on the protest markers, and move the map frequently, the app crashes:
```
Maximum update depth exceeded. 
This can happen when a component repeatedly calls setState inside componentWillUpdate or componentDidUpdate. React limits the number of nested updates to prevent infinite loops.
``` 
Do you experience this too? (I've downgraded leaflet to `1.6.0` locally)

**Comment by oori** [comment]
> Maximum update depth exceeded.

@guytepper Yes, that's exactly the issues I was referring to [above](https://github.com/guytepper/1km.co.il/pull/64#discussion_r502786729).    I thought downgrading to 1.6.0 is a solution to only one problem (marker clicks), but not the dragging. 
This issue exists on your current live site as well ;(      
the issue originates here [OnMoveEnd](https://github.com/guytepper/1km.co.il/blob/ac54eda00fa70fc9b82aa684f68eaa95f45d1f01/src/components/Map/Map.js#L54), but it amplified due to faulty react setState flow, because of `setMapPosition` called too fast (_like fast-duplicate OnMoveEnd calls_).  We should probably move this to a new issue, it's unrelated to this PR.    Anyway, I will check this bug out again later.

**Comment by guytepper** [comment]
@oori once again, thanks a lot and sorry for the delay. I've improved the popup styles & deployed to production. 
Please let us know once Leaflet fix that issue :)
